CollectorUtils/UpdateGNSSSMetadata.py

```diff
@@ -0,0 +1,293 @@
+import argparse
+import sys
+import urllib
+import urllib.parse
+import urllib.request
+import json
+from arcgis.gis import GIS
+
+
+# Parse Command-line arguments
+def parseArguments():
+    global args_parser
+    parser = argparse.ArgumentParser(
+        usage='Add/Update GNSS metadate fields')
+    parser.add_argument('-u', '--username', required=True, type=str, help='Organization username')
+    parser.add_argument('-p', '--password', required=True, type=str, help='Organization password')
+    parser.add_argument('-url', '--url', required=True, type=str, help='Organization url')
+    parser.add_argument('-r', '--remove', default=False, type=bool,
+                        help='Set True if GNSS metadata fields need to be removed')
+    parser.add_argument('itemId', type=str, nargs="+", help='Search string')
+    args_parser = parser.parse_args()
+
+
+# Generate Token based on username and password
+def generateToken():
+    global token
+
+    datapayload = bytes(urllib.parse.urlencode( \
+        {'username': args_parser.username, 'password': args_parser.password, 'referer': args_parser.url, 'f': 'json',
+         'expiration': 60}), 'utf-8')
+    token_response = urllib.request.urlopen(urllib.parse.urljoin(args_parser.url, '/sharing/rest/GenerateToken?'),
+                                            datapayload).read()
+    token = json.loads(token_response)['token']
+    if not token:
+        print('Failed to generate Token .. ')
+        sys.exit(-2)
+
+
+# Search for a item id and add GNSS Metadata fields
+def searchItems_addGNSSMetadataFields():
+    global featureServiceUrl
```

Comment:
Not sure why this is global. I don't see it used anywhere else besides this function.